libraries/AP_RPM/AP_RPM.cpp

```diff
@@ -259,6 +259,40 @@ bool AP_RPM::arming_checks(size_t buflen, char *buffer) const
     return true;
 }
 
+float AP_RPM::get_average_rpm() const
+{
+    float rpm_avg = 0.0f;
+    uint8_t valid_escs = 0;
+
+    // average the rpm of each motor
+    for (uint8_t i = 0; i < num_instances; i++) {
+        float rpm;
+        if (get_rpm(i, rpm)) {
+            rpm_avg += rpm;
+            valid_escs++;
+        }
+    }
+
+    if (valid_escs > 0) {
+        rpm_avg /= valid_escs;
+    }
+
+    return rpm_avg;
+}
+
+uint8_t AP_RPM::get_motor_frequencies_hz(uint8_t nfreqs, float* freqs) const
+{
+    uint8_t valid_rpms = 0;
+    for (uint8_t ii = 0; ii < num_instances && ii < nfreqs; ii++) {
```

Comment:
``ii` as an index is a bit odd in this context. Not that it really matters ;)
```suggestion
    for (uint8_t ii = 0; ii < num_instances && ii < nfreqs; ii++) {
```